### PR TITLE
V4.1.2 no reindex before cfund

### PR DESCRIPTION
**Pull request by aguycalled**
If the CFund db is corrupted or has been removed, the wallet will ask the user to reindex/resync in order to be able to rebuild the CFund DB.

This does not make any sense before the Community Fund is enabled, as it would make every mainnet user to completely resync the chain when v4.1.2 is released.

A new flag is written to the block tree reflecting the current activation state of the CFund, and this is checked on init when validating the DB integrity.

### CODE REVIEW

**Comment by matt-auckland** [comment]
"This does not make any sense before the Community Fund is enabled, as it would make every mainnet user to completely resync the chain when v4.1.2 is released." Nice catch 😛

**Comment by matt-auckland** [comment]
My phone can't deal with these large files so I'll check the code when I get to the office

File: src/base58.cpp [hunk 1]
```diff
@@ -188,9 +188,6 @@ bool CBase58Data::SetString(const std::string& str)
 
 std::string CBase58Data::ToString() const
 {
-//    if(vchData.To == "0x202020")
-//      return "FUND_CONTRIBUTION";
-
```
Comment (matt-auckland) [inline src/base58.cpp]: why don't we need this anymore?

Comment (aguycalled) [reply src/base58.cpp]: its a ghost comment from some internal test i did i forgot to remove. we never needed it.

**Comment by matt-auckland** [comment]
Ah I see

**Pull request merged**